src/util/timing/goblint_timing.mli

```diff
@@ -0,0 +1,3 @@
+(** Profiling with custom hierarchical timed sections. *)
+
+include Goblint_timing_intf.Goblint_timing (** @inline *)
```

Comment:
Any particular reason for this construction?

Reply:
It avoids having to duplicate the type and module type definitions both in `.mli` and `.ml`: https://www.craigfe.io/posts/the-intf-trick.

Reply:
I was aware of what it does (but not that it's something Jane Street does in their packages all the time.
I'm just not sure if it is smart to introduce this here given that we don't use this pattern anywhere else....

Reply:
Ok, then we can start doing now.